vj4/handler/homework.py

```diff
@@ -157,7 +157,7 @@ class HomeworkCodeHandler(base.OperationHandler):
                       file_name='{}.zip'.format(tdoc['title']))
 
 
-@app.route('/homework/{tid}/{pid:-?\d+|\w{24}}', 'homework_detail_problem')
+@app.route('/homework/{tid}/{pid:[A-Z0-9]+}', 'homework_detail_problem')
```

Comment:
Why not allowing lowercase letters and maybe other characters?

Reply:
![image](https://user-images.githubusercontent.com/29992205/57569754-49613080-742c-11e9-8c45-aa64f7210378.png)
It creates conflict.

Reply:
What conflict? Can you explain more?

Reply:
problem 'submit' not found

Reply:
not this route
my fault.
![image](https://user-images.githubusercontent.com/29992205/57569848-e40e3f00-742d-11e9-9ed8-92610bda7ccc.png)
this one.

Reply:
Emmm. Then I think we should not simply support string PIDs without changing the URL scheme. I think depending on URL case sensitivity is not a good idea.

By the way, it would be better to text instead of screenshot for code snippets. It is more friendly for more types of browsers/screen readers. For example:

```python
@app.route(...)
class XxxHandler:
  pass
```

Reply:
Some user may just collected a URL in his browser.
Changing the URL scheme will make then confused as it throws 404 error.

Reply:
Of course we should change the URL in a backward compatible way, and preferably still generate the same URL for numeric/objectid PIDs. This can be achieved by making multiple routes to point to the same handler.

One proposal is to use (optional) named parameter in URL. For example: `/homework/tid=xxx/pid=xxx/scoreboard`. We can also use query strings.

What do you think?

Reply:
Scoreboard for each problem?
so weird...

Reply:
My bad. The example should be `/homework/tid=xxx/pid=xxx/submit`. It is just an example.

Reply:
the submit handler can work properly.
but handlers like 'homework_edit' 'homework_scoreboard' will be handled by 'homework_problem' handler if we enables lowercase pid.